src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -617,17 +616,28 @@ class DateBucketPartitioner(gran: Granularity, intervals: Iterable[Interval]) ex
   }
 }
 
-class DateBucketAndHashPartitioner(gran: Granularity, partMap: Map[(Long, Long), Int], optionalDims : Option[Set[String]] = None)
+class DateBucketAndHashPartitioner(gran: Granularity,
+                                   partMap: Map[(Long, Long), Int],
+                                   optionalDims: Option[Set[String]] = None,
+                                   optionalDimExclusions: Option[Set[String]] = None
+                                  )
   extends Partitioner {
-  val maxTimePerBucket = partMap.groupBy(_._1._1).map(e => e._1 -> e._2.size)
-  val dims = optionalDims.orNull
-
+  lazy val shardLookups = partMap.groupBy(_._1._1)
```

Comment:
you can make this a little bit more explicit by doing something like
`partMap groupBy { case ((bucket, _), _) => bucket }`